src/lib/basic_graph/basic-node.js

```diff
@@ -20,11 +20,14 @@ export default class BasicNode {
   }
 
   _calculateMaxDistanceFromRoot(): integer {
-    if (_.isEmpty(this.parents())) {
-      return 0
-    } else {
-      const maxChildDistanceFromRoot = Math.max(...this.parents().map(c => c.maxDistanceFromRoot))
-      return (maxChildDistanceFromRoot + 1)
+    // We initialize to -1 as you alwasy get one increment for free in the loop below.
+    let distanceFromRoot = -1
+    let ancestors = [this]
+    while (ancestors.length > 0) {
+      distanceFromRoot++
+      ancestors = _.uniq(_.flatten(ancestors.map(e => e.parents())))
```

Comment:
this could be a bit simpler by doing something like ``_(ancestors).uniq.flatten.filter(e => (e.id !==this.id))``